e2e/scenario_test.go

```diff
@@ -901,3 +904,33 @@ func Test_ubuntu2204Wasm(t *testing.T) {
 		},
 	})
 }
+
+func Test_noEchoedSecretsInProvisionLog(t *testing.T) {
+	logPath := "/var/log/azure/cluster-provision.log"
+	secretContent := "**** this secret should not be logged ****"
+	clientPrivateKey := base64.StdEncoding.EncodeToString([]byte("ClientPrivateKey: " + secretContent))
+	spSecret := base64.StdEncoding.EncodeToString([]byte("ServicePrincipalSecret: " + secretContent))
+
+	// Getting the cluster so we can also check for the bootstrap token.
+	cluster, err := ClusterKubenet(newTestCtx(t), t)
```

Comment:
nit: this var name is a bit misleading, it's really the NodeBootstrappingConfiguration we use to bootstrap nodes against the e2e cluster

Reply:
ah, I see you're copying what other tests are doing, we should probably change the naming a bit...

fine to leave as is